### PR TITLE
Fix bemtree nested promises problem

### PR DESCRIPTION
**Pull request by apsavin**
For #1079

### CODE REVIEW

**Comment by apsavin** [comment]
@tadatuta please take a look. By the way, I don't understand what's wrong with code style.

**Comment by tadatuta** [comment]
> I don't understand what's wrong with code style

https://travis-ci.org/bem/bem-core/builds/69351751
1. There shouldn't be space before `?` mark.
2. var statements should be joined:

``` js
var a = 1,
    b = 2;
```

**Comment by tadatuta** [comment]
cc @narqo @veged

**Comment by narqo** [comment]
@apsavin could you show a usecase which is failing now. I'm not sure I've understood what exactly this path fixes?

**Comment by tadatuta** [comment]
@narqo  https://nda.ya.ru/3QodZh

File: common.blocks/i-bem/i-bem.bemtree [hunk 1]
```diff
@@ -92,16 +92,23 @@ BEMContext.prototype.doAsync = function doAsync(fn) {
...
 };
 
-// Wrap xjst's apply and export our own
-var oldApply = exports.apply;
+/**
+ * waits for all, even nested, promises inside bemtree
+ * @param ctx
+ * @returns {Promise}
+ */
+var waitForPromises = function (ctx) {
```
Comment (veged) [inline common.blocks/i-bem/i-bem.bemtree]: it's better to do `function waitForPromises(ctx) {` and avoid to change 95-96 lines

**Comment by veged** [comment]
CodeStyle: also should't have spaces after `function`

**Comment by veged** [comment]
LGTM

**Comment by apsavin** [comment]
Code style fixed

**Comment by veged** [comment]
thnx!
could you also port it to `v3`, please?

**Comment by apsavin** [comment]
Of course, I'll do it soon.

**Pull request merged**